src/main/scala/io/iohk/ethereum/domain/BlockHeader.scala

```diff
@@ -102,24 +128,42 @@ object BlockHeader {
   }
 
   implicit class BlockheaderEncodableDec(val rlpEncodeable: RLPEncodeable) extends AnyVal {
```

Comment:
Not directly related to this PR, but I'm not fond of this naming: `BlockheaderEncodableDec` is by types symmetrical to `BlockHeaderEnc` (`RLPEncodable => BlockHeader` vs `BlockHeader => RLPEncodable`), so it should be named `BlockHeaderDec`.

What is currently  `BlockheaderDec` should be renamed to `BlockHeaderByteArrayDec`, especially that it's only used in tests.

Note: `Blockheader*` -> `BlockHeader*`